File: iosApp/iosApp/Features/Diary/notes/components/AddNoteItem.swift
//
//  AddNoteItem.swift
//  iosApp
//
//  Created by velkonost on 03.10.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct AddNoteItem: View {
    
    let paddingBottom: Int
    let createGoalClick: () -> Void
    let createNoteClick: () -> Void
    
    init(paddingBottom: Int = 90, createGoalClick: @escaping () -> Void, createNoteClick: @escaping () -> Void) {
        self.paddingBottom = paddingBottom
        self.createGoalClick = createGoalClick
        self.createNoteClick = createNoteClick
    }
    
    
    var body: some View {
        VStack {
            Spacer()
            HStack {
                Spacer()
                ExpandableButtonPanel(
                    paddingBottom: paddingBottom,
                    primaryItem: ExpandableButtonItem(
                        icon: SharedR.images().ic_plus.toUIImage()!,
                        iconExpanded: SharedR.images().ic_arrow_back.toUIImage()!,
                        label: SharedR.strings().cancel.desc().localized()
                    ),
                    secondaryItems: [
                        ExpandableButtonItem(
                            icon: SharedR.images().ic_goal.toUIImage()!,
                            iconExpanded: nil,
                            label: SharedR.strings().create_goal_button.desc().localized()
                        ) {
                            createGoalClick()
                        },
                        ExpandableButtonItem(
                            icon: SharedR.images().ic_note.toUIImage()!,
                            iconExpanded: nil,
                            label: SharedR.strings().create_note_button.desc().localized()
                        ) {
                            createNoteClick()
                        }
                    ]
                )
            }
        }
    }
}

File: androidApp/core/compose/src/main/kotlin/com/velkonost/getbetter/core/compose/components/notelist/AddNoteItem.kt
package com.velkonost.getbetter.core.compose.components.notelist

import androidx.compose.foundation.layout.BoxScope
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.Dp
import com.velkonost.getbetter.core.compose.components.ExpandableButtonItem
import com.velkonost.getbetter.core.compose.components.ExpandableButtonPanel
import com.velkonost.getbetter.core.compose.theme.Dimen.DP_140
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.painterResource
import dev.icerock.moko.resources.compose.stringResource

@Composable
fun BoxScope.AddNoteItem(
    modifier: Modifier = Modifier,
    paddingBottom: Dp = DP_140,
    createGoalClick: () -> Unit,
    createNoteClick: () -> Unit
) {
    ExpandableButtonPanel(
        modifier = modifier,
        paddingBottom = paddingBottom,
        primaryItem = ExpandableButtonItem(
            title = stringResource(resource = SharedR.strings.cancel),
            icon = painterResource(imageResource = SharedR.images.ic_plus),
            iconExpanded = painterResource(imageResource = SharedR.images.ic_arrow_back)
        ),
        items = listOf(
            ExpandableButtonItem(
                title = stringResource(resource = SharedR.strings.create_goal_button),
                icon = painterResource(imageResource = SharedR.images.ic_goal)
            ) { createGoalClick.invoke() },

            ExpandableButtonItem(
                title = stringResource(resource = SharedR.strings.create_note_button),
                icon = painterResource(imageResource = SharedR.images.ic_note)
            ) { createNoteClick.invoke() }
        ),
    )
}